cmd/otel-allocator/target/discovery_test.go

```diff
@@ -52,17 +53,27 @@ func TestDiscovery(t *testing.T) {
 			want: []string{"prom.domain:9004", "prom.domain:9005", "promfile.domain:1001", "promfile.domain:3000"},
 		},
 	}
-	manager := NewManager(context.Background(), ctrl.Log.WithName("test"), gokitlog.NewNopLogger(), nil)
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	d := discovery.NewManager(ctx, gokitlog.NewNopLogger())
+	manager := NewDiscoverer(ctrl.Log.WithName("test"), d, nil)
 	defer close(manager.close)
+	defer cancelFunc()
```

Comment:
nit (and out of curiosity): does order of defers matter? and should this be called directly below L56 (could L57 and L58 ever cause a panic where the cancel function would not be called)